### ISSUE TITLE
generation time has increased when upgrading to v2.14.0

### ISSUE THREAD

**Issue by HasanAlqaisi** [simolus3/moor#2770]
**Describe the bug**
Hi,
I've upgraded to drift v2.14 and I noticed that running build_runner took much more time than the previous version.
According to [this issue](https://github.com/simolus3/drift/issues/2533) I optimized the generation time before with build.yaml.
But when upgraded it seems that the generator doesn't respect the build.yaml options because it took the same time as before.

**Comment by kuhnroyal** [comment]
We are also experiencing a huge increase from ~50sec to ~3-4min

**Comment by Vedsaga** [comment]
Same here taking alot of time earlier with just freezed package time was not much more then 30-40 sec... 

btw tried, 
```yaml
targets:
  $default:
    builders:
      drift_dev:
        generate_for:
           - "lib/config/app_db"
 ```
 
 for some reason it's not generating all the files in this why is that? It seems `Companion` Object never gets generated... when removing the this build.ymal file then again things works..

**Comment by simolus3** [comment]
The examples part of the repository are too small to reliably tell slowdowns apart from timing noise in `build_runner`, but a bisect points to d770c16d8d9790479fde55ef8d870c05beadcd3b.

I have reverted that commit on the `investigate-build-performance` branch, could you try adding a dependency override to confirm whether that fixes the problem (perhaps run a clean build with `dart run build_runner clean && dart run build_runner build -d`)?

```yaml
dependency_overrides:
  drift_dev:
    git:
      url: https://github.com/simolus3/drift.git
      ref: investigate-build-performance
      path: drift_dev
```

**Comment by HasanAlqaisi** [comment]
> The examples part of the repository are too small to reliably tell slowdowns apart from timing noise in `build_runner`, but a bisect points to [d770c16](https://github.com/simolus3/drift/commit/d770c16d8d9790479fde55ef8d870c05beadcd3b).
> 
> I have reverted that commit on the `investigate-build-performance` branch, could you try adding a dependency override to confirm whether that fixes the problem (perhaps run a clean build with `dart run build_runner clean && dart run build_runner build -d`)?
> 
> ```yaml
> dependency_overrides:
>   drift_dev:
>     git:
>       url: https://github.com/simolus3/drift.git
>       ref: investigate-build-performance
>       path: drift_dev
> ```

I've tried that and I can confirm that it fixes the problem

**Comment by lyio** [comment]
With 2.14.0:
`[INFO] Succeeded after 5m 24s with 5832 outputs (7405 actions)`

With your investigation branch:
`[INFO] Succeeded after 54.1s with 1509 outputs (3082 actions)`

**Comment by Vedsaga** [comment]
can also confirms that things are faster with ` investigation branch` ...

----
Btw, I wonder, why this `build.ymal` file not able to generate class for Appdatabase 


```yaml
targets:
  $default:
    builders:
      drift_dev:
        generate_for:
           - "lib/config/app_db/**"
  ```
  

As can be seen in the image below that all the files related to app-db is here itself in the ling/config folder...  
![image](https://github.com/simolus3/drift/assets/59083528/f32db9b1-2a0c-442d-9f23-12600de67f35)


Generating getting error when having custom `build.ymal` file, however after deleting that things works fine...

````
Built build_runner:build_runner.
[INFO] Generating build script completed, took 397ms
[INFO] Precompiling build script... completed, took 8.3s
[INFO] Setting up file watchers completed, took 10ms
[INFO] Waiting for all file watchers to be ready completed, took 182ms
[INFO] Building new asset graph completed, took 1.5s
[INFO] Checking for unexpected pre-existing outputs. completed, took 14ms
[INFO] Generating SDK summary completed, took 6.3s
[WARNING] json_serializable on lib/modules/auth/model/user_preference_data.dart:
You are missing a required dependency on json_annotation in the "dependencies" section of your pubspec with a lower bound of at least "4.8.1".
[WARNING] drift_dev on lib/config/app_db/app_database.dart:
line 14, column 7 of package:mytrailpals/config/app_db/app_database.dart: The referenced element, AppwriteVerificationTokenTable, is not understood by drift.
   ╷
14 │ class AppDatabase extends _$AppDatabase {
   │       ^^^^^^^^^^^
   ╵
[WARNING] drift_dev on lib/config/app_db/app_database.dart:
line 14, column 7 of package:mytrailpals/config/app_db/app_database.dart: The referenced element, AuthTable, is not understood by drift.
   ╷
14 │ class AppDatabase extends _$AppDatabase {
   │       ^^^^^^^^^^^
   ╵
[WARNING] drift_dev on lib/config/app_db/query/appwrite_verification_token_table_query.dart:
line 8, column 7 of package:mytrailpals/config/app_db/query/appwrite_verification_token_table_query.dart: The referenced element, AppwriteVerificationTokenTable, is not understood by drift.
  ╷
8 │ class AppwriteVerificationTokenTableQuery extends DatabaseAccessor<AppDatabase>
  │       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
[WARNING] drift_dev on lib/config/app_db/query/auth_table_query.dart:
line 8, column 7 of package:mytrailpals/config/app_db/query/auth_table_query.dart: The referenced element, AuthTable, is not understood by drift.
  ╷
8 │ class AuthTableQuery extends DatabaseAccessor<AppDatabase>
  │       ^^^^^^^^^^^^^^
  ╵
[INFO] Running build completed, took 38.7s
[INFO] Caching finalized dependency graph completed, took 183ms
[INFO] Succeeded after 38.9s with 168 outputs (1409 actions)
```

**Comment by simolus3** [comment]
Thanks for the feedback, this should be fixed with the `drift_dev: ^2.14.1` release I've just published. Please let me know if you still run into issues with that.

> As can be seen in the image below that all the files related to app-db is here itself in the ling/config folder...

Yeah, that's weird - it should work since the tables itself are also matched by the `generate_for`. Are there any files under `.dart_tool/build/generated/<your package name>/lib/config/app_db/model`?

**Comment by HasanAlqaisi** [comment]
> Thanks for the feedback, this should be fixed with the `drift_dev: ^2.14.1` release I've just published. Please let me know if you still run into issues with that.
> 
> > As can be seen in the image below that all the files related to app-db is here itself in the ling/config folder...
> 
> Yeah, that's weird - it should work since the tables itself are also matched by the `generate_for`. Are there any files under `.dart_tool/build/generated/<your package name>/lib/config/app_db/model`?

It seems that the issue is fixed in `drift_dev: ^2.14.1`
Thank you for your time!
